When specific nuget version is needed by a package it is the chocolatey version that is used

I have a problem similar of the one in ticket 194.
I have received this message during installation of a software with choco install:

    The 'Microsoft.NETCore.Platforms 3.0.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.11.0.0'.

I have looked at ticket 194 but I need more help to solve it again.

So please tell me how to solve it.
Thanks,
Mario

> 2019-10-15 23:35:33,627 11980 [INFO ] - Chocolatey v0.10.15
2019-10-15 23:35:33,643 11980 [DEBUG] - Chocolatey is running on Windows v 10.0.18362.0
2019-10-15 23:35:33,643 11980 [DEBUG] - Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
2019-10-15 23:35:33,643 11980 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
2019-10-15 23:35:33,674 11980 [DEBUG] - Command line: "C:\ProgramData\chocolatey\choco.exe" install NETStandard.Library -dv --log-file='c:\choco_output.log'
2019-10-15 23:35:33,674 11980 [DEBUG] - Received arguments: install NETStandard.Library -dv --log-file='c:\choco_output.log'
2019-10-15 23:35:33,737 11980 [DEBUG] - RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
2019-10-15 23:35:33,752 11980 [DEBUG] - Sending message 'PreRunMessage' out if there are subscribers...
2019-10-15 23:35:33,768 11980 [DEBUG] - [Pending] Removing all pending packages that should not be considered installed...
2019-10-15 23:35:33,846 11980 [DEBUG] - Performing validation checks.
2019-10-15 23:35:33,861 11980 [DEBUG] - Global Configuration Validation Checks:
2019-10-15 23:35:33,861 11980 [DEBUG] -  - Package Exit Code / Exit On Reboot = Checked
2019-10-15 23:35:33,861 11980 [DEBUG] - System State Validation Checks:
2019-10-15 23:35:33,861 11980 [DEBUG] -  Reboot Requirement Checks:
2019-10-15 23:35:33,877 11980 [DEBUG] -  - Pending Computer Rename = Checked
2019-10-15 23:35:33,877 11980 [DEBUG] -  - Pending Component Based Servicing = Checked
2019-10-15 23:35:33,877 11980 [DEBUG] -  - Pending Windows Auto Update = Checked
2019-10-15 23:35:33,877 11980 [DEBUG] -  - Pending File Rename Operations = Ignored
2019-10-15 23:35:33,877 11980 [DEBUG] -  - Pending Windows Package Installer = Checked
2019-10-15 23:35:33,893 11980 [DEBUG] -  - Pending Windows Package Installer SysWow64 = Checked
2019-10-15 23:35:33,893 11980 [INFO ] - 2 validations performed. 2 success(es), 0 warning(s), and 0 error(s).
2019-10-15 23:35:33,893 11980 [DEBUG] - The source 'https://chocolatey.org/api/v2/;https://systemrepo.giammar.it/repository/nuget-group/' evaluated to a 'normal' source type
2019-10-15 23:35:33,893 11980 [DEBUG] -
NOTE: Hiding sensitive configuration data! Please double and triple
 check to be sure no sensitive data is shown, especially if copying
 output to a gist for review.
2019-10-15 23:35:33,924 11980 [DEBUG] - Configuration: CommandName='install'|
CacheLocation='C:\Users\Mario Giammarco\AppData\Local\Temp\chocolatey'|
ContainsLegacyPackageInstalls='True'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='https://chocolatey.org/api/v2/;https://systemrepo.giammar.it/repository/nuget-group/'|

SourceType='normal'|Debug='True'|Verbose='True'|Trace='False'|
Force='False'|Noop='False'|HelpRequested='False'|
UnsuccessfulParsing='False'|RegularOutput='True'|QuietOutput='False'|
PromptForConfirmation='True'|AcceptLicense='False'|
AllowUnofficialBuild='False'|
AdditionalLogFileLocation='c:\choco_output.log'|
Input='NETStandard.Library'|AllVersions='False'|
SkipPackageInstallProvider='False'|PackageNames='NETStandard.Library'|
Prerelease='False'|ForceX86='False'|OverrideArguments='False'|
NotSilent='False'|ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
AllowMultipleVersions='False'|AllowDowngrade='False'|
ForceDependencies='False'|Information.PlatformType='Windows'|
Information.PlatformVersion='10.0.18362.0'|
Information.PlatformName='Windows 10'|
Information.ChocolateyVersion='0.10.15.0'|
Information.ChocolateyProductVersion='0.10.15'|
Information.FullName='choco, Version=0.10.15.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='Mario Giammarco'|
Information.UserDomainName='DESKTOP-1SMR5TO'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='True'|
Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|Information.LicenseType='Foss'|
Information.CurrentDirectory='C:\WINDOWS\system32'|
Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseEnhancedExitCodes='False'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='False'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.SkipPackageUpgradesWhenNotInstalled='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ExitOnRebootDetected='False'|
Features.LogValidationResultsOnWarnings='True'|
Features.UsePackageRepositoryOptimizations='True'|
Features.ScriptsCheckLastExitCode='False'|
ListCommand.LocalOnly='False'|
ListCommand.IdOnly='False'|ListCommand.IncludeRegistryPrograms='False'|
ListCommand.PageSize='25'|ListCommand.Exact='False'|
ListCommand.ByIdOnly='False'|ListCommand.ByTagOnly='False'|
ListCommand.IdStartsWith='False'|ListCommand.OrderByPopularity='False'|
ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='unknown'|
ApiKeyCommand.Remove='False'|PinCommand.Command='unknown'|
OutdatedCommand.IgnorePinned='False'|Proxy.BypassOnLocal='True'|
2019-10-15 23:35:33,955 11980 [DEBUG] - _ Chocolatey:ChocolateyInstallCommand - Normal Run Mode _
2019-10-15 23:35:33,971 11980 [INFO ] - Installing the following packages:
2019-10-15 23:35:33,971 11980 [INFO ] - NETStandard.Library
2019-10-15 23:35:33,986 11980 [INFO ] - By installing you accept licenses for the packages.
2019-10-15 23:35:34,127 11980 [DEBUG] - Using 'https://chocolatey.org/api/v2/'.
2019-10-15 23:35:34,783 11980 [DEBUG] - - Supports prereleases? 'True'.
2019-10-15 23:35:34,799 11980 [DEBUG] - - Is ServiceBased? 'True'.
2019-10-15 23:35:35,627 11980 [DEBUG] - Using 'https://systemrepo.giammar.it/repository/nuget-group/'.
2019-10-15 23:35:36,080 11980 [DEBUG] - - Supports prereleases? 'True'.
2019-10-15 23:35:36,096 11980 [DEBUG] - - Is ServiceBased? 'True'.
2019-10-15 23:35:37,235 11980 [DEBUG] - Package 'netstandard.library' found on source 'https://systemrepo.giammar.it/repository/nuget-group/'
2019-10-15 23:35:37,341 11980 [INFO ] - [NuGet] Attempting to resolve dependency 'Microsoft.NETCore.Platforms (= 1.1.0)'.
2019-10-15 23:35:37,565 11980 [ERROR] - NETStandard.Library not installed. An error occurred during installation:
 The 'Microsoft.NETCore.Platforms 3.0.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.11.0.0'.
2019-10-15 23:35:37,582 11980 [INFO ] - NETStandard.Library package files install completed. Performing other installation steps.
2019-10-15 23:35:37,588 11980 [DEBUG] - No package information as package is null.
2019-10-15 23:35:37,615 11980 [DEBUG] - No package information to save as package is null.
2019-10-15 23:35:37,623 11980 [DEBUG] - Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers...
2019-10-15 23:35:37,629 11980 [ERROR] - The install of NETStandard.Library was NOT successful.
2019-10-15 23:35:37,636 11980 [ERROR] - NETStandard.Library not installed. An error occurred during installation:
 The 'Microsoft.NETCore.Platforms 3.0.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.11.0.0'.
2019-10-15 23:35:37,660 11980 [WARN ] -
Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2019-10-15 23:35:37,668 11980 [INFO ] -
2019-10-15 23:35:37,672 11980 [ERROR] - Failures
2019-10-15 23:35:37,676 11980 [ERROR] -  - NETStandard.Library (exited 1) - NETStandard.Library not installed. An error occurred during installation:
 The 'Microsoft.NETCore.Platforms 3.0.0' package requires NuGet client version '2.12' or above, but the current NuGet version is '2.11.0.0'.
2019-10-15 23:35:37,682 11980 [WARN ] -
Did you know the proceeds of Pro (and some proceeds from other
 licensed editions) go into bettering the community infrastructure?
 Your support ensures an active community, keeps Chocolatey tip top,
 plus it nets you some awesome features!
 https://chocolatey.org/compare
2019-10-15 23:35:37,695 11980 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2019-10-15 23:35:37,770 11980 [DEBUG] - Exiting with 1

--- a/src/chocolatey.tests.integration/context/unsupportedelements/1.0.0/unsupportedelements.nuspec
+++ b/src/chocolatey.tests.integration/context/unsupportedelements/1.0.0/unsupportedelements.nuspec
@@ -1,6 +1,6 @@
 <?xml version="1.0"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
-  <metadata>
+  <metadata minClientVersion="100.0.0">
     <id>unsupportedelements</id>
     <version>1.0.0</version>
     <title>unsupportedelements</title>

--- a/src/chocolatey.tests.integration/context/unsupportedelements/1.1.0/unsupportedelements.nuspec
+++ b/src/chocolatey.tests.integration/context/unsupportedelements/1.1.0/unsupportedelements.nuspec
@@ -1,6 +1,6 @@
 <?xml version="1.0"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
-  <metadata>
+  <metadata minClientVersion="100.0.0">
     <id>unsupportedelements</id>
     <version>1.1.0</version>
     <title>unsupportedelements</title>

--- a/src/chocolatey.tests.integration/scenarios/PackScenarios.cs
+++ b/src/chocolatey.tests.integration/scenarios/PackScenarios.cs
@@ -499,6 +499,27 @@ namespace chocolatey.tests.integration.scenarios
             }
         }
 
+        public class when_packing_with_min_client_version : ScenariosBase
+        {
+            protected override string ExpectedNuspecVersion => "0.1.0";
+            protected override string ExpectedSubDirectory => "PackageOutput";
+
+            // This high version is to ensure that pack does not throw, even if the min client version is well
+            // above both the Chocolatey and NuGet assembly versions.
+            private string MinClientVersion = "100.0.0";
+
+            public override void Because()
+            {
+                MockLogger.reset();
+                Service.pack_run(Configuration);
+            }
+
+            protected override string GetNuspecContent()
+            {
+                return NuspecContentWithFormatableMinClientVersion.format_with(MinClientVersion);
+            }
+        }
+
         private const string NuspecContent = @"<?xml version=""1.0"" encoding=""utf-8""?>
 <package xmlns=""http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd"">
   <metadata>
@@ -900,6 +921,29 @@ namespace chocolatey.tests.integration.scenarios
     <id>test-package</id>
     <title>Test Package</title>
     <version>{0}</version>
+    <authors>package author</authors>
+    <owners>package owner</owners>
+    <summary>A brief summary</summary>
+    <description>A big description</description>
+    <tags>test admin</tags>
+    <copyright></copyright>
+    <licenseUrl>http://apache.org/2</licenseUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <releaseNotes></releaseNotes>
+    <dependencies>
+      <dependency id=""chocolatey-core.extension"" />
+    </dependencies>
+  </metadata>
+  <files>
+  </files>
+</package>";
+
+    private const string NuspecContentWithFormatableMinClientVersion = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<package xmlns=""http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd"">
+  <metadata minClientVersion=""{0}"">
+    <id>test-package</id>
+    <title>Test Package</title>
+    <version>0.1.0</version>
     <authors>package author</authors>
     <owners>package owner</owners>
     <summary>A brief summary</summary>

--- a/src/chocolatey.tests.integration/scenarios/PackScenarios.cs
+++ b/src/chocolatey.tests.integration/scenarios/PackScenarios.cs
@@ -499,24 +499,24 @@ namespace chocolatey.tests.integration.scenarios
             }
         }
 
-        public class when_packing_with_min_client_version : ScenariosBase
+        public class When_packing_with_min_client_version : ScenariosBase
         {
             protected override string ExpectedNuspecVersion => "0.1.0";
             protected override string ExpectedSubDirectory => "PackageOutput";
 
             // This high version is to ensure that pack does not throw, even if the min client version is well
             // above both the Chocolatey and NuGet assembly versions.
-            private string MinClientVersion = "100.0.0";
+            private string _minClientVersion = "100.0.0";
 
             public override void Because()
             {
-                MockLogger.reset();
-                Service.pack_run(Configuration);
+                MockLogger.Reset();
+                Service.Pack(Configuration);
             }
 
             protected override string GetNuspecContent()
             {
-                return NuspecContentWithFormatableMinClientVersion.format_with(MinClientVersion);
+                return NuspecContentWithFormatableMinClientVersion.FormatWith(_minClientVersion);
             }
         }
 

--- a/src/chocolatey.tests/infrastructure.app/nuget/NugetCommonSpecs.cs
+++ b/src/chocolatey.tests/infrastructure.app/nuget/NugetCommonSpecs.cs
@@ -19,6 +19,7 @@ namespace chocolatey.tests.infrastructure.app.nuget
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using chocolatey.infrastructure.app;
     using chocolatey.infrastructure.app.configuration;
     using chocolatey.infrastructure.app.nuget;
     using chocolatey.infrastructure.filesystem;
@@ -160,6 +161,22 @@ namespace chocolatey.tests.infrastructure.app.nuget
                     .ShouldEqual(("file:" + source).Replace("\\", "/"));
                 packageRepositories.First().PackageSource.IsLocal.ShouldBeTrue();
                 packageRepositories.First().PackageSource.SourceUri.IsUnc.ShouldBeTrue();
+            }
+
+            [Fact]
+            public void Should_set_user_agent_string()
+            {
+                Context();
+                var source = "https://community.chocolatey.org/api/v2/";
+                configuration.Sources = source;
+                configuration.Information.ChocolateyProductVersion = "vNext";
+
+                because();
+
+                // Change this when the NuGet version is updated.
+                string nugetClientVersion = "6.4.1";
+                string expectedUserAgentString = "{0}/{1} via NuGet Client/{2}".FormatWith(ApplicationParameters.UserAgent, configuration.Information.ChocolateyProductVersion, nugetClientVersion);
+                UserAgent.UserAgentString.ShouldStartWith(expectedUserAgentString);
             }
         }
     }

